pkgs/tools/typesetting/tex/texlive/default.nix

```diff
@@ -144,6 +304,59 @@ let
     xzcat "$tlpdb" | sed -rn -f "$tl2nix" | uniq > "$out"
   '';
 
+  # build tlType == "bin" containers based on `binfiles` in TLPDB
+  # see UPGRADING.md for how to keep the list of shebangs up to date
+  # TODO: add extraBuildInputs (?) for adding PATH entries to texlive.combine wrappers
+  mkPkgBin = pname: version: run: { binfiles, scriptsFolder ? pname, postFixup ? "", formats ? [], ... }@args:
+    runCommand "texlive-${pname}.bin-${version}"
+      ( {
+          # metadata for texlive.combine
+          passthru = {
+            inherit pname version;
+            tlType = "bin";
+          };
+          # shebang interpreters
+          # TODO: package SNOBOL4
+          buildInputs = [ bash perl python3 bin.luatex ruby tk ];
+          run = run.outPath or "";
+          inherit scriptsFolder;
+        }
+      )
+      # look for scripts
+      # the explicit list of extensions avoid non-scripts such as $binname.cmd, $binname.jar, $binname.pm
+      # the order is relevant: $binname.sh is preferred to other $binname.*
+      (''
+        mkdir -p "$out/bin"
+        for binname in "${builtins.concatStringsSep "\" \"" binfiles}" ; do
+          target="$out/bin/$binname"
```

Comment:
this approach should also fix #223923 as the wrapper created in `combine` now calls the correctly named `albatross` symlink.

Reply:
indeed! `albatross` works on my machine now